src/test/java/com/amazon/opendistroforelasticsearch/sql/antlr/semantic/types/GenericTypeTest.java

```diff
@@ -34,17 +35,17 @@
 
     @Test
     public void passNumberArgToLogShouldReturnNumber() {
-        assertEquals(NUMBER, LOG.construct(singletonList(NUMBER)));
+        assertEquals(DOUBLE, LOG.construct(singletonList(NUMBER)));
     }
 
     @Test
     public void passIntegerArgToLogShouldReturnInteger() {
-        assertEquals(INTEGER, LOG.construct(singletonList(INTEGER)));
+        assertEquals(DOUBLE, LOG.construct(singletonList(INTEGER)));
     }
 
     @Test
     public void passLongArgToLogShouldReturnLong() {
-        assertEquals(LONG, LOG.construct(singletonList(LONG)));
+        assertEquals(DOUBLE, LOG.construct(singletonList(LONG)));
     }
```

Comment:
is there a need for test cases going the other way around (`e.g assertEquals(LONG, LOG.construct(singletonList(DOUBLE))`) ?

Reply:
Done. Change the test case name to passLongArgToLogShouldReturnDouble.